xarray/plot/plot.py

```diff
@@ -29,19 +29,27 @@
 
 
 def _infer_line_data(darray, x, y, hue):
-    error_msg = "must be either None or one of ({:s})".format(
-        ", ".join([repr(dd) for dd in darray.dims])
-    )
+    error_msg = "must be a dimension, coordinate, MultiIndex level name or None"
     ndims = len(darray.dims)
 
-    if x is not None and x not in darray.dims and x not in darray.coords:
-        raise ValueError("x " + error_msg)
+    if (
+        x is not None
+        and x not in darray.dims
```

Comment:
can we make `valid_xy = set(darray.dims) + set(darray.coords) + set(darray._level_coords)` and then print those values in the error message. It would make the if conditions on x and y pretty clean too.

Could do the same below too. 

Is it possible to refactor out the `x` and `y` checking?